### PR TITLE
Rework of notifications UI

### PR DESCRIPTION
**Pull request by DanielW1987**
would be good if you can find some more time to test and question how it works. It turned out a bit different than on the last screenshot.

### CODE REVIEW

**Comment by DanielW1987** [comment]
[Re-Layout of Notification Config](https://trello.com/c/2nKVuKuh/355-re-layout-of-notification-config)

**Comment by DanielW1987** [comment]
I don't have the linting error on my local machine.

File: webapp/src/main/java/rocks/metaldetector/service/notification/config/TelegramConfigServiceImpl.java [hunk 1]
```diff
@@ -79,10 +84,35 @@ public int generateRegistrationId() {
...
 
     telegramConfig.setRegistrationId(registrationId);
     telegramConfigRepository.save(telegramConfig);
     return registrationId;
   }
+
+  private NotificationConfigEntity getOrCreateNotificationConfig(AbstractUserEntity user) {
+    Optional<NotificationConfigEntity> notificationConfigOrEmpty = notificationConfigRepository.findByUserAndChannel(user, TELEGRAM);
+    if (notificationConfigOrEmpty.isPresent()) {
+      return notificationConfigOrEmpty.get();
```
Comment (hedgehog1833) [inline webapp/src/main/java/rocks/metaldetector/service/notification/config/TelegramConfigServiceImpl.java:100]: At the latest in this line the naming ...OrEmpty is a bit weird, maybe we stick with ...Optional? 😉

File: webapp/src/main/resources/templates/frontend/notification-settings.html [hunk 2]
```diff
@@ -10,63 +11,129 @@
...
+                                            <label class="custom-control-label" for="email-2-weekly-rb">2-weekly</label>
+                                        </div>
+                                        <div class="custom-control custom-radio mb-2">
+                                            <input type="radio" id="email-4-weekly-rb" name="email-regular-notification-frequency" class="custom-control-input">
+                                            <label class="custom-control-label" for="email-4-weekly-rb">4-weekly</label>
+                                        </div>
+                                    </div>
+                                </div>
+                                <div class="form-group mb-4">
+                                    <label class="font-weight-bold" id="extra-notifications-label">Extra Notifications</label><br />
```
Comment (hedgehog1833) [inline webapp/src/main/resources/templates/frontend/notification-settings.html:54]:
```suggestion
                                    <label class="font-weight-bold" id="extra-notifications-label">Extra notifications</label><br />
```

File: webapp/src/main/resources/templates/frontend/notification-settings.html [hunk 3]
```diff
@@ -10,63 +11,129 @@
...
+                                                    <label class="custom-control-label" for="telegram-2-weekly-rb">2-weekly</label>
+                                                </div>
+                                                <div class="custom-control custom-radio mb-2">
+                                                    <input type="radio" id="telegram-4-weekly-rb" name="telegram-regular-notification-frequency" class="custom-control-input">
+                                                    <label class="custom-control-label" for="telegram-4-weekly-rb">4-weekly</label>
+                                                </div>
+                                            </div>
+                                        </div>
+                                        <div class="form-group mb-4">
+                                            <label class="font-weight-bold">Extra Notifications</label><br />
```
Comment (hedgehog1833) [inline webapp/src/main/resources/templates/frontend/notification-settings.html:115]:
```suggestion
                                            <label class="font-weight-bold">Extra notifications</label><br />
```

File: webapp/src/main/resources/templates/frontend/notification-settings.html [hunk 4]
```diff
@@ -10,63 +11,129 @@
...
+                                            </div>
+                                            <div class="custom-control custom-switch">
+                                                <input type="checkbox" class="custom-control-input" id="telegram-announcement-date-notification-toggle">
+                                                <label class="custom-control-label" for="telegram-announcement-date-notification-toggle">Notification on announcement date</label>
+                                            </div>
+                                        </div>
+                                    </fieldset>
+                                    <div class="form-group">
+                                        <button class="btn btn-outline-danger btn-lg disabled" id="deactivate-telegram-button">
+                                            <span class="material-icons">delete</span> Delete bot configuration
```
Comment (hedgehog1833) [inline webapp/src/main/resources/templates/frontend/notification-settings.html:128]: I'm not sure if the user know's what the bot is here.... Maybe "Delete Telegram configuration" or just "Delete configuration"?

File: webapp/src/main/resources/templates/frontend/notification-settings.html [hunk 5]
```diff
@@ -10,63 +11,129 @@
...
-                <p>
-                    You can get email notifications about recent or upcoming releases at a frequency of 2 or 4 weeks.
-                    These emails are always sent on Sundays.
-                </p>
-                <div class="form-group">
-                    <div class="custom-control custom-switch">
-                        <input type="checkbox" class="custom-control-input" id="notification-toggle">
-                        <label class="custom-control-label" for="notification-toggle">Regular notifications</label>
+            <p>
+                You can receive notifications about current or upcoming releases at a frequency of 2 or 4 weeks. The notifications are
```
Comment (hedgehog1833) [inline webapp/src/main/resources/templates/frontend/notification-settings.html:17]:
```suggestion
                You can receive notifications about current or upcoming releases at a frequency of two or four weeks. These notifications are
```
The numbers looks better i think. The "These" makes it clear that only the period notifications come on Sundays, the others can can be different of course.

**Review by hedgehog1833 — COMMENTED** [review]
Looks really great! Just some small things

**Review by hedgehog1833 — APPROVED** [review]
👍

**Pull request merged**